### PR TITLE
Feature/local notification

### LINKED ISSUE #14: iOS Refactor - local notification
# Refactor

- [x] codebase refactoring     
      1. use Singleton or not (need to handle race condition)
      2. DI relationship (Coordinator or ViewModel)

- [x] unit test

Link: https://github.com/No-57/SCSH_3C_iOS/pull/13

### PR DESCRIPTION
**Pull request by Vince-Ku**
**Remote Push Notification Flow Diagram**.
Please note that I'm using a free Apple Developer Account, so I can only demonstrate using local notification.

Screenshot 2023-09-05 at 12 47 52 PM

### CODE REVIEW

**Comment by shadow3x3x3** [comment]
nit: Let's set up our git config email to align it to your github email.

For your local git config, run the following command:

```bash
git config --global user.email "xxx@gmail.com"
```

Push any commits after `git config` and you should see commit authors linked to your github account.

**Comment by Vince-Ku** [comment]
> nit: Let's set up our git config email to align it to your github email.
> 
> For your local git config, run the following command:
> 
> ```shell
> git config --global user.email "xxx@gmail.com"
> ```
> 
> Push any commits after `git config` and you should see commit authors linked to your github account.

Oh, just realized I'm still on my old company's account! XD
Thanks !!

File: SCSH_3C_iOS/SCSH_3C_iOS/Ecosystem/Notification/NotificationFactory.swift [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import UserNotifications
+
+protocol NotificationFactoryType {
+    func generate(identifier: String, title: String, body: String, triggerInterval: TimeInterval) -> UNNotificationRequest
+}
+
+extension NotificationFactoryType {
+    func generate(identifier: String = UUID().uuidString, title: String, body: String, triggerInterval: TimeInterval = 1) -> UNNotificationRequest {
+        generate(identifier: identifier, title: title, body: body, triggerInterval: triggerInterval)
```
Comment (shadow3x3x3) [inline SCSH_3C_iOS/SCSH_3C_iOS/Ecosystem/Notification/NotificationFactory.swift:16]: Is this an idiom of Swift extensions, calling a function with the same name within a function?

Comment (Vince-Ku) [reply SCSH_3C_iOS/SCSH_3C_iOS/Ecosystem/Notification/NotificationFactory.swift:16]: In Swift, this is how we implement `Overloading`.
In reality, the function signatures are distinct, as you can see, one of them includes a default value of `UUID().uuidString`.

Comment (Vince-Ku) [reply SCSH_3C_iOS/SCSH_3C_iOS/Ecosystem/Notification/NotificationFactory.swift:16]: Regarding whether it's an idiom or not?

I'd say... maybe? after all, the Swift compiler DOES warn you if you cause any infinite recursion.

I think that is your concern?

Comment (shadow3x3x3) [reply SCSH_3C_iOS/SCSH_3C_iOS/Ecosystem/Notification/NotificationFactory.swift:16]: Oh.... It makes sense if this is overloading. I am OK with it.

**Review by shadow3x3x3 — APPROVED** [review]

**Pull request merged**